src/hooks/useSelectedWallet.js

```diff
@@ -0,0 +1,41 @@
+// @flow
+
+import { type EdgeCurrencyWallet } from 'edge-core-js'
+
+import { useWatchAccount } from '../hooks/useWatch.js'
+import { useSelector } from '../types/reactRedux.js'
+
+export type SelectedWallet = {
+  currencyCode: string,
+  tokenId?: string,
+  wallet: EdgeCurrencyWallet
+}
+
+export function useSelectedWallet(): SelectedWallet | void {
+  const walletId = useSelector(state => state.ui.wallets.selectedWalletId)
+  const currencyCode = useSelector(state => state.ui.wallets.selectedCurrencyCode)
+
+  // Grab the wallet from the account:
+  const account = useSelector(state => state.core.account)
+  const currencyWallets = useWatchAccount(account, 'currencyWallets')
+  const wallet = currencyWallets[walletId]
+  if (wallet == null) return
+
+  // We cannot use any hooks after the return statement,
+  // but we don't need to worry about `allTokens` being stale,
+  // because the selected token must exist before being selected,
+  // so the selector above will force us to render in any case:
+  const { allTokens } = wallet.currencyConfig
+  const tokenId =
```

Comment:
You can use the already existing 'getTokenId' method - https://github.com/EdgeApp/edge-react-gui/blob/d8788689a2b6d46fd277750543eece24c64a287b/src/util/CurrencyInfoHelpers.js#L99